account_move_line_create_journal_entry/models/account_move_line.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2025 Quartile (https://www.quartile.co)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, models
+from odoo.exceptions import UserError
+
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    def action_create_journal_entry(self):
+        if len(set(self.mapped("partner_id.id"))) > 1:
+            raise UserError(_("All selected records must have the same partner."))
+        if any(line.move_id.move_type != "in_invoice" for line in self):
```

Comment:
```suggestion
        if any(line.move_type != "in_invoice" for line in self):
```
Shouldn't we be supporting all in_xxx and out_xxx types (at least all in_xxx types)?